src/main/java/gameframework/motion/MoveStrategy.java

```diff
@@ -1,5 +1,9 @@
 package gameframework.motion;
 
 public interface MoveStrategy {
-	SpeedVector getSpeedVector();
+	public SpeedVector getSpeedVector();
+	
+	public int getSpeed();
+	
+	public void setSpeed(int speed);
```

Comment:
Except the DefaultImpl which returns a null SpeedVector, I think every MoveStrategy can have a settable speed.
Anyway, if you want I can delete setSpeed from the interface